src/huggingface_hub/inference/_client.py

```diff
@@ -1870,6 +1870,11 @@ def _resolve_url(self, model: Optional[str] = None, task: Optional[str] = None)
                     "You must specify at least a model (repo_id or URL) or a task, either when instantiating"
                     " `InferenceClient` or when making a request."
                 )
+            logger.info(
+                f"Using recommended model {model} for task {task}. Note that it is"
+                f" encouraged to explicitly set `model='{model}'` as the recommended"
+                " models list might get updated without prior notice."
+            )
```

Comment:
This line has to be added just after `model = _get_recommended_model(task)` so that `model` is correctly set in the message.

Reply:
Oh shoot, nice catch.  yeah I pasted in the wrong spot

Reply:
Implemented